src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return String(this[0]);
+    }
```

Comment:
can we process one element inside the cycle (start it from 0 index) instead of create separate check for the case with 1 element?